File: documentation-examples-scala/src/scala/dev/typr/foundations/docs/oracle/DomainType.scala
package dev.typr.foundations.docs.oracle
import dev.typr.foundationssc.*
import dev.typr.foundationssc.data.*



@SuppressWarnings(Array("unused"))
object DomainType:
  //start
  // Wrapper type
  case class EmployeeId(value: Long)

  // Create OracleType from NUMBER
  val empIdType: OracleType[EmployeeId] =
    OracleTypes.numberLong.transform(EmployeeId.apply, _.value)
  //stop

File: documentation-examples-java/src/java/dev/typr/foundations/docs/oracle/DomainType.java
package dev.typr.foundations.docs.oracle;

import dev.typr.foundations.OracleType;
import dev.typr.foundations.OracleTypes;

@SuppressWarnings("unused")
public class DomainType {
    //start
    // Wrapper type
    public record EmployeeId(Long value) {}

    // Create OracleType from NUMBER
    OracleType<EmployeeId> empIdType =
        OracleTypes.numberLong.transform(EmployeeId::new, EmployeeId::value);
    //stop
}
